### PR TITLE
cli: add ability to autoconfirm sending spading data

### PR DESCRIPTION
**Pull request by midgleyc**
If you have a lot of spading data, confirming each time is a bit annoying. This adds a command to confirm everything pre-emptively.

### CODE REVIEW

**Comment by jaadams5** [comment]
Not about the patch but the use case.  I don't see anything in mafia that is actually collecting spading data so this change is presumably to make a script easier to use?  @gausie 's Excavator has code that bypasses the need for confirmation.  Not sure it is still being maintained, though.

**Comment by midgleyc** [comment]
It's for excavator's 70s volcano quest collection, which isn't automatically submitted. Might be an excavator bug if it's supposed to be.

File: src/net/sourceforge/kolmafia/textui/command/SubmitSpadeDataCommand.java
Comment (jaadams5) [inline src/net/sourceforge/kolmafia/textui/command/SubmitSpadeDataCommand.java:44]: The previous behavior was the spade command always cleared data whether it had been sent or not.  Since we are here is that still what we want?

File: src/net/sourceforge/kolmafia/session/LoginManager.java
Comment (jaadams5) [inline src/net/sourceforge/kolmafia/session/LoginManager.java:219]: Please type `spade' to examine and optionally submit the data or `spade autoconfirm` to submit all of the spaded data.  Either way the data will be deleted whether shared or not.

Not sure that is really better but it is clearer.

**Review by jaadams5 — DISMISSED** [review]
Consider the always delete behaviour and the wording of the "nag" but those are suggestions not requirements.

File: src/net/sourceforge/kolmafia/textui/command/SubmitSpadeDataCommand.java
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/textui/command/SubmitSpadeDataCommand.java:44]: I also thought that was weird but it's outside the scope of this change, in my opinion: I just left it as it was.

File: src/net/sourceforge/kolmafia/session/LoginManager.java
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/session/LoginManager.java]: updated, thanks

**Comment by midgleyc** [comment]
Looks like `Dismiss stale pull request approvals when new commits are pushed` is enabled on this repository: I've changed the wording as you requested and it's dismissed your review (:o).

**Comment by jaadams5** [comment]
> Looks like `Dismiss stale pull request approvals when new commits are pushed` is enabled on this repository: I've changed the wording as you requested and it's dismissed your review (:o).

That is the expected behavior.  :-)

**Comment by jaadams5** [comment]
> It's for excavator's 70s volcano quest collection, which isn't automatically submitted. Might be an excavator bug if it's supposed to be.

My experience is that the automatic submission never worked and I had to add something to my logout file.  See https://kolmafia.us/threads/excavator-gausies-spading-script.25076/

Maybe the best solution is to fix Excavator?


I also note that deleting all data, sent or not, via space is at least documented, if not necessarily desirable, behaviour.

**Review by jaadams5 — APPROVED** [review]
I'm OK with the changes.  Discussion about changing mafia or excavator should be resolved before merging :-)

**Comment by jaadams5** [comment]
@midgleyc 

if (get_property("spadingData") != ""){
	cli_execute("try; ash import excavator; flush_spading_data()");
}

seems to be an existing solution.

Does it make sense to close this without committing?

**Comment by midgleyc** [comment]
Possibly. There's a bit of a halfway house in terms of support at the moment, with Mafia having the "spade" command, but it only really being used by excavator.

There's a workaround for the problem I was having (as you've posted, thanks), but it's not very discoverable, and the problem would re-arise if a non-excavator spading script came into use.

**Comment by jaadams5** [comment]
> Possibly. There's a bit of a halfway house in terms of support at the moment, with Mafia having the "spade" command, but it only really being used by excavator.
> 
> There's a workaround for the problem I was having (as you've posted, thanks), but it's not very discoverable, and the problem would re-arise if a non-excavator spading script came into use.

Bingo.  Either this gets approved or the existing spade command gets deprecated, IMO.  Since only one of those is non-controversial...

**Pull request merged**
